hypothesis-python/src/hypothesis/extra/numpy.py

```diff
@@ -1312,7 +1312,10 @@ def do_draw(self, data):
         else:
             while result[-1:] == [slice(None, None)] and data.draw(st.integers(0, 7)):
                 result.pop()
-        return tuple(result)
+        if len(result) == 1 and data.draw(st.booleans()):
+            return result[0]
+        else:
+            return tuple(result)
 
 
 @st.defines_strategy
```

Comment:
We'll also want to update the docstring for `basic_indices()`, to document that bare elements may be generated in place of length-one tuples.  e.g.:

>       ``shape``-shaped array will produce either a scalar or a shared-memory view.
>     + When a length-one tuple would be generated, this strategy may instead return
>     + the element which will index the first axis, e.g. `5` instead of `(5,)`.

Reply:
Agree, it's better to document that.